### PR TITLE
Show central package url when pushing already existing package to the central

### PR DESCRIPTION
**Pull request by pramodya1994**
```
ballerina: package 'pramodya/xa:0.1.12' already exists in remote repository(https://central.ballerina.io/pramodya/xa). build and push after updating the version in the Ballerina.toml.
```

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/29499

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by NipunaRanasinghe — COMMENTED** [review]
Do we need to add a negative test case for the above scenario?

**Comment by pramodya1994** [comment]
> Do we need to add a negative test case for the above scenario?

Currently, we have added central-related tests to the distribution repo. We can have negative tests there. We can add an issue for this.

**Review by hemikak — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java [hunk 1]
```diff
@@ -54,6 +54,8 @@
     private static final String STAGING_URL = "https://api.staging-central.ballerina.io/2.0/registry";
     private static final String DEV_URL = "https://api.dev-central.ballerina.io/2.0/registry";
 
+    private static final String BALLERINA_CENTRAL_URL = "https://central.ballerina.io";
+
```
Comment (hemikak) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java:58]:
```suggestion
    private static final String BALLERINA_CENTRAL_URL = "https://central.ballerina.io";
    private static final String BALLERINA_CENTRAL_STAGING_URL = "https://staging-central.ballerina.io";
    private static final String BALLERINA_CENTRAL_DEV_URL = "https://dev-central.ballerina.io";
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java [hunk 2]
```diff
@@ -151,6 +153,10 @@ public static String getRemoteRepoURL() {
         return PRODUCTION_URL;
     }
 
+    public static String getCentralPackageURL(String org, String pkgName) {
+        return BALLERINA_CENTRAL_URL + "/" + org + "/" + pkgName;
```
Comment (hemikak) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java:157]:
```suggestion
        if (SET_BALLERINA_STAGE_CENTRAL) {
            return BALLERINA_CENTRAL_STAGING_URL + "/" + org + "/" + pkgName;
        } else if (SET_BALLERINA_DEV_CENTRAL) {
            return BALLERINA_CENTRAL_DEV_URL + "/" + org + "/" + pkgName;
        }
        return BALLERINA_CENTRAL_PRODUCTION_URL + "/" + org + "/" + pkgName;
```

**Comment by hemikak** [comment]
Added code suggestions. Please check.

**Comment by pramodya1994** [comment]
> Added code suggestions. Please check.

Resolved in 2aae7e8

**Review by hemikak — DISMISSED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java [hunk 3]
```diff
@@ -210,8 +210,9 @@ private static Path validateBala(BuildProject project, CentralAPIClient client)
                     + pkgAsDependency.version().toString();
             throw new ProjectException(
                     "package '" + pkg + "' already exists in " + "remote repository("
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:212]:
```suggestion
                    "package '" + pkg + "' already exists in " + "remote repository: "
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java [hunk 4]
```diff
@@ -54,6 +54,10 @@
     private static final String STAGING_URL = "https://api.staging-central.ballerina.io/2.0/registry";
     private static final String DEV_URL = "https://api.dev-central.ballerina.io/2.0/registry";
 
+    private static final String BALLERINA_CENTRAL_PRODUCTION_URL = "https://central.ballerina.io";
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java:57]: Shall we move these to CentralUtils?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java [hunk 5]
```diff
@@ -151,6 +155,22 @@ public static String getRemoteRepoURL() {
...
     }
 
+    /**
+     * Get the central package URL.
+     *
+     * @param org     package org
+     * @param pkgName package name
+     * @return central package URL
+     */
+    public static String getCentralPackageURL(String org, String pkgName) {
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/util/RepoUtils.java:165]: Related to https://github.com/ballerina-platform/ballerina-lang/pull/30182/#r621129979

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java (hunk 3 above)
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:212]:
```suggestion
                    "package '" + pkg + "' already exists in " + "remote repository ("
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java [hunk 6]
```diff
@@ -210,8 +210,9 @@ private static Path validateBala(BuildProject project, CentralAPIClient client)
...
-                            + getRemoteRepoURL() + "). build and push after "
-                            + "updating the version in the Ballerina.toml.");
+                            + getCentralPackageURL(project.currentPackage().packageOrg().value(),
+                                                   project.currentPackage().packageName().value())
+                            + "). build and push after updating the version in the Ballerina.toml.");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java:215]:
```suggestion
                            + ". build and push after updating the version in the Ballerina.toml.");
```

**Review by azinneera — APPROVED** [review]

**Pull request merged**